itests/raft_messagesigner_test.go

```diff
@@ -0,0 +1,238 @@
+
+	blockTime := 5 * time.Millisecond
+
+	pkey0, _ := generatePrivKey()
+	pkey1, _ := generatePrivKey()
+	pkey2, _ := generatePrivKey()
+
+	initPeerSet := []peer.ID{pkey0.PeerID, pkey1.PeerID, pkey2.PeerID}
+
+	raftOps := kit.ConstructorOpts(
+		node.Override(new(*gorpc.Client), modules.NewRPCClient),
+		node.Override(new(*config.ClusterRaftConfig), func() *config.ClusterRaftConfig {
+			cfg := config.DefaultClusterRaftConfig()
+			cfg.InitPeerset = initPeerSet
+			return cfg
+		}),
+		node.Override(new(*consensus.Consensus), consensus.NewConsensusWithRPCClient(false)),
+		node.Override(new(*messagesigner.MessageSignerConsensus), messagesigner.NewMessageSignerConsensus),
+		node.Override(new(messagesigner.MsgSigner), func(ms *messagesigner.MessageSignerConsensus) *messagesigner.MessageSignerConsensus { return ms }),
+		node.Override(new(*modules.RPCHandler), modules.NewRPCHandler),
+		node.Override(node.RPCServer, modules.NewRPCServer),
+	)
+	//raftOps := kit.ConstructorOpts()
+
+	ens := kit.NewEnsemble(t).FullNode(node0, raftOps).FullNode(node1, raftOps).FullNode(node2, raftOps)
+	node0.AssignPrivKey(pkey0)
+	node1.AssignPrivKey(pkey1)
+	node2.AssignPrivKey(pkey2)
+
+	ens.MinerEnroll(miner, node0, kit.WithAllSubsystems())
+	ens.Start()
+
+	// Import miner wallet to all nodes
+	addr0, err := node0.WalletImport(ctx, &miner.OwnerKey.KeyInfo)
+	require.NoError(t, err)
+	addr1, err := node1.WalletImport(ctx, &miner.OwnerKey.KeyInfo)
+	require.NoError(t, err)
+	addr2, err := node2.WalletImport(ctx, &miner.OwnerKey.KeyInfo)
+	require.NoError(t, err)
+
+	fmt.Println(addr0, addr1, addr2)
+
+	ens.InterconnectAll()
+
+	ens.AddInactiveMiner(miner)
+	ens.Start()
+
+	ens.InterconnectAll().BeginMining(blockTime)
+
+	return ens
+}
+
+func TestRaftState(t *testing.T) {
+
+	kit.QuietMiningLogs()
+	ctx := context.Background()
+
+	var (
+		node0 kit.TestFullNode
+		node1 kit.TestFullNode
+		node2 kit.TestFullNode
+		miner kit.TestMiner
+	)
+
+	setup(ctx, t, &node0, &node1, &node2, &miner)
+
+	fmt.Println(node0.WalletList(context.Background()))
+	fmt.Println(node1.WalletList(context.Background()))
+	fmt.Println(node2.WalletList(context.Background()))
+
+	bal, err := node0.WalletBalance(ctx, node0.DefaultKey.Address)
+	require.NoError(t, err)
+
+	msgHalfBal := &types.Message{
+		From:  miner.OwnerKey.Address,
+		To:    node0.DefaultKey.Address,
+		Value: big.Div(bal, big.NewInt(2)),
+	}
+	mu := uuid.New()
+	smHalfBal, err := node0.MpoolPushMessage(ctx, msgHalfBal, &api.MessageSendSpec{
+		MsgUuid: mu,
+	})
+	require.NoError(t, err)
+	mLookup, err := node0.StateWaitMsg(ctx, smHalfBal.Cid(), 3, api.LookbackNoLimit, true)
+	require.NoError(t, err)
+	require.Equal(t, exitcode.Ok, mLookup.Receipt.ExitCode)
+
+	rstate0 := getRaftState(ctx, t, &node0)
+	rstate1 := getRaftState(ctx, t, &node1)
+	rstate2 := getRaftState(ctx, t, &node2)
+
+	require.True(t, reflect.DeepEqual(rstate0, rstate1))
```

Comment:
I believe you can use `require.EqualValues(t, rstate0, rstate1)`, which uses reflect.DeepEqual under the hood (but also prints a nice diff when values don't match)

Reply:
done